packages/model/src/entities/inline-fragments/FormattingInlineNode/types/IntersectType.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Types of interaction between two inline fragments with intersect ranges
+ */
+export enum IntersectType {
+  /**
+   * If two fragments of one tool intersect - merge into one fragment
+   */
+  Extend = 'extend',
+
+  /**
+   * If two new fragment intersect with existing - remove existing with new one
+   */
+  Replace = 'replace',
+
+  /**
+   * If two fragments of one tool intersect - treat them as two different fragments
+   */
+  Separate = 'separate'
```

Comment:
I don't really like that name, I think it's not descriptive enough. Maybe something like `LeaveBoth` or `None` as there's no intersection in that case. Happy to discuss other ideas